crates/hir_ty/src/infer/pat.rs

```diff
@@ -111,20 +119,29 @@ impl<'a> InferenceContext<'a> {
         let expected = expected;
 
         let ty = match &body[pat] {
-            Pat::Tuple { ref args, .. } => {
+            &Pat::Tuple { ref args, ellipsis } => {
                 let expectations = match expected.as_tuple() {
                     Some(parameters) => &*parameters.0,
                     _ => &[],
                 };
-                let expectations_iter = expectations.iter().chain(repeat(&Ty::Unknown));
 
-                let inner_tys = args
-                    .iter()
-                    .zip(expectations_iter)
-                    .map(|(&pat, ty)| self.infer_pat(pat, ty, default_bm))
-                    .collect();
+                let (pre, post) = match ellipsis {
+                    Some(idx) => args.split_at(idx),
+                    None => (&args[..], &[][..]),
+                };
+                let uncovered_range = pre.len()..expectations.len().saturating_sub(post.len());
```

Comment:
I don't see this being used except to call `len()` on it, so maybe just calculate the length directly?